frontend-react/src/components/MessageTracker/MessageReceivers.tsx

```diff
@@ -1,62 +1,479 @@
+    const handleColHeaderClick = () => {
+        if (!isCurrentlyActiveColumn) {
+            // Reset active column and sort order on new column click
+            setActiveColumn(columnHeaderTitle);
+            setActiveColumnSortOrder(FilterOptionsEnum.None);
+        }
+
+        // Explicitly set the proceeding sort order
+        switch (true) {
+            case activeColumnSortOrder === FilterOptionsEnum.None:
+                setActiveColumnSortOrder(FilterOptionsEnum.ASC);
+                break;
+            case activeColumnSortOrder === FilterOptionsEnum.ASC:
+                setActiveColumnSortOrder(FilterOptionsEnum.DESC);
+                break;
+            case activeColumnSortOrder === FilterOptionsEnum.DESC:
+                setActiveColumnSortOrder(FilterOptionsEnum.None);
+                break;
+        }
+    };
+    return (
+        <th
+            className={
+                isCurrentlyActiveColumn &&
+                activeColumnSortOrder !== FilterOptionsEnum.None
+                    ? "active-col-header"
+                    : ""
+            }
+            onClick={handleColHeaderClick}
+            rowSpan={rowSpan}
+        >
+            <div className="column-header-title-container">
+                <p>{columnHeaderTitle}</p>
+                {isCurrentlyActiveColumn ? filterIcon : <Icon.SortArrow />}
+            </div>
+        </th>
+    );
+};
+
+const MessageReceiversRow = ({
+    receiver,
+    activeColumn,
+    activeColumnSortOrder,
+    setModalText,
+    modalRef,
+}: MessageReceiversRowProps) => {
+    // We highlight the entire active column so within the row,
+    // we need to know which column is active
+    const checkForActiveColumn = (colName: string) =>
+        colName === activeColumn &&
+        activeColumnSortOrder !== FilterOptionsEnum.None
+            ? "active-col-td"
+            : "";
+    const uppercaseMain = receiver.Main.toUpperCase();
+    return (
+        <tr>
+            <td className={checkForActiveColumn(ColumnDataEnum.Name)}>
+                {receiver.Name}
+            </td>
+            <td className={checkForActiveColumn(ColumnDataEnum.Service)}>
+                {receiver.Service}
+            </td>
+            <td className={checkForActiveColumn(ColumnDataEnum.Date)}>
+                {receiver.Date}
+            </td>
+            <td
+                className={`message-receiver-break-word ${checkForActiveColumn(
+                    ColumnDataEnum.ReportId
+                )}`}
+                onClick={() => {
+                    setModalText({
+                        title: `${ColumnDataEnum.ReportId}:`,
+                        body: receiver.ReportId,
+                    });
+                    modalRef?.current?.toggleModal();
+                }}
+            >
+                {receiver.ReportId}
+            </td>
+            <td className={checkForActiveColumn(ColumnDataEnum.Main)}>
+                <p
```

Comment:
Might be nice to pull out any custom/non-verbatim table cell logic into separate components.  This could be specified on the top level when the Table component is rendered like:

`[null, null, null, StatusBadge, null, ...]`

Reply:
Added to our discussion on what features to add for a genericized table component: https://app.zenhub.com/workspaces/experience-607d9d5e68b95200150fec37/issues/gh/cdcgov/prime-reportstream/8280